### PR TITLE
set_attenuation: adding checker based on config data

### PR DESCRIPTION
**Pull request by fkwasowi**
Adding attenuation checking based on the frame_fmt from the config of the module instead of calling
sink->stream.frame_fmt, because sending IPC_SET_ATTENUATION before calling module_prepare results in an error, because at this stage the sink values are not yet set (sink->stream.frame_fmt is zero)

### CODE REVIEW

**Review by pblaszko — APPROVED** [review]

**Review by dnikodem — APPROVED** [review]

**Review by dabekjakub — APPROVED** [review]

File: src/audio/copier/copier.c [hunk 1]
```diff
@@ -1591,14 +1589,18 @@ static int set_attenuation(struct comp_dev *dev, uint32_t data_offset, const cha
 		comp_err(dev, "attenuation %d is out of range", attenuation);
 		return -EINVAL;
 	}
+	uint32_t __sparse_cache valid_fmt, frame_fmt;
```
Comment (RanderWang) [inline src/audio/copier/copier.c:1592]: can we move the declaration to the beginning of this function ?

**Review by RanderWang — APPROVED** [review]
one concern

File: src/audio/copier/copier.c (hunk 1 above)
Comment (fkwasowi) [reply src/audio/copier/copier.c:1592]: SOF CI

**Comment by fkwasowi** [comment]
SOF CI

File: src/audio/copier/copier.c (hunk 1 above)
Comment (fkwasowi) [reply src/audio/copier/copier.c]: Done

**Pull request merged**